--- origsrc/mplayer-export-2010-10-20/configure	2010-10-15 05:45:19.000000000 -0500
+++ src/mplayer-export-2010-10-20/configure	2010-10-19 23:35:17.053146400 -0500
@@ -7677,7 +7677,7 @@ int main(void) {
   return exitvar;
 }
 EOF
-    cc_check -lXext && _xshape=yes
+    cc_check -lXext -lX11 && _xshape=yes
   fi
   if test "$_xshape" = yes ; then
     def_xshape='#define CONFIG_XSHAPE 1'
@@ -7953,7 +7953,7 @@ if test "$_dvdnav" = yes ; then
     inputmodules="dvdnav(internal) $inputmodules"
   else
     extra_cflags="$extra_cflags $($_dvdnavconfig --cflags)"
-    extra_ldflags="$extra_ldflags $($_dvdnavconfig --minilibs)"
+    extra_ldflags="$($_dvdnavconfig --minilibs) $extra_ldflags"
     inputmodules="dvdnav $inputmodules"
   fi
 else
