--- origsrc/mplayer-export-2010-10-20/configure	2010-10-19 23:35:52.106407500 -0500
+++ src/mplayer-export-2010-10-20/configure	2010-10-19 23:39:46.080427500 -0500
@@ -1507,7 +1507,12 @@ elif linux ; then
   _ranlib='true'
 fi
 
-if win32 ; then
+if cygwin ; then
+  _exesuf=".exe"
+  _pe_executable=yes
+fi
+
+if mingw32 ; then
   _exesuf=".exe"
   extra_cflags="$extra_cflags -fno-common"
   # -lwinmm is always needed for osdep/timer-win2.c
@@ -1517,9 +1522,6 @@ if win32 ; then
   _priority=yes
   def_dos_paths="#define HAVE_DOS_PATHS 1"
   def_priority="#define CONFIG_PRIORITY 1"
-fi
-
-if mingw32 ; then
   _getch=getch2-win.c
   _need_shmem=no
 fi
@@ -1687,7 +1689,7 @@ if test "$_runtime_cpudetection" = no ;
 
 # Cygwin has /proc/cpuinfo, but only supports Intel CPUs
 # FIXME: Remove the cygwin check once AMD CPUs are supported
-if test -r /proc/cpuinfo && ! cygwin; then
+if test -r /proc/cpuinfo; then
   # Linux with /proc mounted, extract CPU information from it
   _cpuinfo="cat /proc/cpuinfo"
 elif test -r /compat/linux/proc/cpuinfo && ! x86 ; then
@@ -4595,7 +4597,7 @@ else
   novomodules="cvidix $novomodules"
 fi
 
-if test "$_vidix" = yes && win32; then
+if test "$_vidix" = yes && mingw32; then
   winvidix=yes
   vomodules="winvidix $vomodules"
   libs_mplayer="$libs_mplayer -lgdi32"
@@ -5814,7 +5816,7 @@ if test "$_vcd" = auto; then
   _vcd=no
   if linux || freebsd || netbsd || openbsd || dragonfly || bsdos || darwin || sunos || os2; then
     _vcd=yes
-  elif mingw32; then
+  elif win32; then
   header_check ddk/ntddcdrm.h && _vcd=yes
   fi
 fi
@@ -5905,7 +5907,7 @@ if test "$_libdvdcss_internal" = yes ; t
     def_dvd_darwin='#define DARWIN_DVD_IOCTL'
     extra_ldflags="$extra_ldflags -framework IOKit -framework Carbon"
   elif cygwin ; then
-    cflags_libdvdcss="-DSYS_CYGWIN -DWIN32"
+    cflags_libdvdcss="-DSYS_CYGWIN"
   elif beos ; then
     cflags_libdvdcss="-DSYS_BEOS"
   elif os2 ; then
@@ -7650,7 +7652,7 @@ if test "$_gui" = yes ; then
   fi
   test "$_freetype" = no && test "$_bitmap_font" = no && \
     die "The GUI requires either FreeType or bitmap font support."
-if ! win32 ; then
+if ! mingw32 ; then
   _gui_gtk=yes
   test "$_x11" != yes && die "X11 support required for GUI compilation."
 
--- origsrc/mplayer-export-2010-10-20/libaf/af_ladspa.c	2010-06-25 11:53:27.000000000 -0500
+++ src/mplayer-export-2010-10-20/libaf/af_ladspa.c	2010-10-19 23:39:46.096027600 -0500
@@ -296,7 +296,7 @@ static void* mydlopen(const char *filena
     size_t filenamelen;
     void *result = NULL;
 
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
     /* For Windows there's only absolute path support.
      * If you have a Windows machine, feel free to fix this.
      * (path separator, shared objects extension, et cetera). */
--- origsrc/mplayer-export-2010-10-20/libdvdread4/bswap.h	2008-10-09 17:04:24.000000000 -0500
+++ src/mplayer-export-2010-10-20/libdvdread4/bswap.h	2010-10-19 23:41:43.704689500 -0500
@@ -37,7 +37,7 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__linux__) || defined(__GLIBC__)
+#if defined(__linux__) || defined(__GLIBC__) || defined(__CYGWIN__)
 #include <byteswap.h>
 #define B2N_16(x) x = bswap_16(x)
 #define B2N_32(x) x = bswap_32(x)
--- origsrc/mplayer-export-2010-10-20/libvo/video_out.c	2010-09-20 02:33:53.000000000 -0500
+++ src/mplayer-export-2010-10-20/libvo/video_out.c	2010-10-20 00:13:13.991284800 -0500
@@ -259,7 +259,7 @@ const vo_functions_t* const video_out_dr
 #ifdef CONFIG_X11
         &video_out_xvidix,
 #endif
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
         &video_out_winvidix,
 #endif
         &video_out_cvidix,
--- origsrc/mplayer-export-2010-10-20/mpcommon.c	2010-10-17 10:25:38.000000000 -0500
+++ src/mplayer-export-2010-10-20/mpcommon.c	2010-10-20 00:18:00.250832000 -0500
@@ -375,7 +375,7 @@ void common_preinit(void)
  */
 static void sanitize_os(void)
 {
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
     HMODULE kernel32 = GetModuleHandle("Kernel32.dll");
     BOOL WINAPI (*setDEP)(DWORD) = NULL;
     BOOL WINAPI (*setDllDir)(LPCTSTR) = NULL;
--- origsrc/mplayer-export-2010-10-20/mplayer.c	2010-10-17 10:23:33.000000000 -0500
+++ src/mplayer-export-2010-10-20/mplayer.c	2010-10-19 23:39:46.111627600 -0500
@@ -34,7 +34,7 @@
 #include <sys/time.h>
 #include <sys/types.h>
 
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
 #define _UWIN 1  /*disable Non-underscored versions of non-ANSI functions as otherwise int eof would conflict with eof()*/
 #include <windows.h>
 #endif
@@ -719,7 +719,7 @@ void exit_player_with_rc(enum exit_reaso
 
   if (mpctx->user_muted && !mpctx->edl_muted) mixer_mute(&mpctx->mixer);
   uninit_player(INITIALIZED_ALL);
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
   timeEndPeriod(1);
 #endif
 #ifdef CONFIG_X11
@@ -2795,7 +2795,7 @@ int gui_no_filename=0;
     }
 
   print_version("MPlayer");
-#if (defined(__MINGW32__) || defined(__CYGWIN__)) && defined(CONFIG_GUI)
+#if defined(__MINGW32__) && defined(CONFIG_GUI)
     void *runningmplayer = FindWindow("MPlayer GUI for Windows", "MPlayer for Windows");
     if (runningmplayer && filename && use_gui) {
         COPYDATASTRUCT csData;
@@ -2818,7 +2818,7 @@ int gui_no_filename=0;
       use_gui=0;
     }
 #else
-#if !defined(__MINGW32__) && !defined(__CYGWIN__)
+#if !defined(__MINGW32__)
     if(use_gui && !vo_init()){
       mp_msg(MSGT_CPLAYER,MSGL_WARN,MSGTR_GuiNeedsX);
       use_gui=0;
@@ -3202,7 +3202,7 @@ while (player_idle_mode && !filename) {
       char *buf = strdup(filename), *psub;
       char *pdot = strrchr(buf, '.');
       char *pslash = strrchr(buf, '/');
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
       if (!pslash) pslash = strrchr(buf, '\\');
 #endif
       if (pdot && (!pslash || pdot > pslash))
@@ -3213,7 +3213,7 @@ while (player_idle_mode && !filename) {
           char *bname;
           int l;
           bname = strrchr(buf,'/');
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
           if(!bname) bname = strrchr(buf,'\\');
 #endif
           if(bname) bname++;
--- origsrc/mplayer-export-2010-10-20/path.c	2010-03-23 13:01:00.000000000 -0500
+++ src/mplayer-export-2010-10-20/path.c	2010-10-19 23:39:46.096027600 -0500
@@ -65,7 +65,7 @@ char *get_path(const char *filename){
 	if ((homedir = getenv("MPLAYER_HOME")) != NULL)
 		config_dir = "";
 	else if ((homedir = getenv("HOME")) == NULL)
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
 	/* Hack to get fonts etc. loaded outside of Cygwin environment. */
 	{
 		int i,imax=0;
--- origsrc/mplayer-export-2010-10-20/stream/stream_vcd.c	2010-06-21 03:38:23.000000000 -0500
+++ src/mplayer-export-2010-10-20/stream/stream_vcd.c	2010-10-19 23:39:46.127227600 -0500
@@ -161,7 +161,7 @@ static int open_s(stream_t *stream,int m
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
 
-#if defined(__MINGW32__) || defined(__CYGWIN__)
+#if defined(__MINGW32__)
   device[4] = p->device[0];
   /* open() can't be used for devices so do it the complicated way */
   hd = CreateFile(device, GENERIC_READ, FILE_SHARE_READ, NULL,
