libdvdnavmini is static only (explanation in the configure file)

--- origsrc/mplayer/configure	2009-03-02 16:56:56.000000000 -0600
+++ src/mplayer/configure	2009-03-03 02:03:27.508515800 -0600
@@ -7679,7 +7678,7 @@ int main(void) {
   return exitvar;
 }
 EOF
-    cc_check -lXext && _xshape=yes
+    cc_check -lXext -lX11 && _xshape=yes
   fi
   if test "$_xshape" = yes ; then
     def_xshape='#define CONFIG_XSHAPE 1'
@@ -8000,7 +7999,7 @@ if test "$_dvdnav" = yes ; then
     _inputmodules="dvdnav(internal) $_inputmodules"
   else
     _inc_extra="$_inc_extra `$_dvdnavconfig --cflags`"
-    _ld_extra="$_ld_extra `$_dvdnavconfig --minilibs`"
+    _ld_extra="`$_dvdnavconfig --minilibs` $_ld_extra"
     _inputmodules="dvdnav $_inputmodules"
   fi
 else
