http://www.freebsd.org/cgi/query-pr.cgi?pr=130966

--- origsrc/mplayer/gui/wm/ws.c	2009-03-04 01:25:35.000000000 -0600
+++ src/mplayer/gui/wm/ws.c	2009-03-05 23:57:04.712500000 -0600
@@ -1172,6 +1172,7 @@ void wsCreateImage( wsTWindow * win,int 
    win->xImage->data=win->Shminfo.shmaddr;
    win->Shminfo.readOnly=0;
    XShmAttach( wsDisplay,&win->Shminfo );
+   XSync(wsDisplay, False);
    shmctl( win->Shminfo.shmid,IPC_RMID,0 );
   }
   else
